packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -92,6 +93,13 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
       apiRef.current.state,
     );
 
+    React.useEffect(() => {
+      if (props.state != null && gridState !== props.state) {
```

Comment:
I would recommend that we determine the controlled vs non-controlled state during the initial mount. It's what we do for the internal states of the other components with this line: https://github.com/mui-org/material-ui/blob/81ee3c5b72531a4513900962b5aa292a6878e466/packages/material-ui-utils/src/useControlled.js#L6 (there is a warning throw when this is not respected). We can change it later on. Once, we see somebody find a but related to it.
 
In the current logic, it's done at each render.

Reply:
Well it's done in an effect so it means that it will run if one of the dependency changes

> > What do you mean?
> 
> * Looking closer, it might be handled, I think that I will wait to see the documentation and the tests to be sure.
> * I assume we miss a prop to set the default state, should it be handled in a different pull request, later on?
> * On a related, the [controllable API](https://react-table.tanstack.com/docs/faq#how-can-i-manually-control-the-table-state) of react-table is puzzling me. I have never seen such a pattern before. I don't understand the incentive for having a prop that expects a state function (oldState => newState). Strange. It seems to have been introduced in [tannerlinsley/react-table#1704](https://github.com/tannerlinsley/react-table/issues/1704). I don't see any completing argument in the thread. So, I think that we can continue in the direction we have started, and ignore it.

We'll do the default in another PR